test/unittest/statistics/dds/StatisticsQosTests.cpp

```diff
@@ -67,6 +104,211 @@ TEST(StatisticsQosTests, StatisticsDataReaderQosTest)
+            statistics_publisher_impl->lookup_datawriter(subscription_throughput_name);
+    ASSERT_EQ(subscription_throughput_writer, nullptr);
+
+    // TEST FOR NEW PUBLIC METHOD: enable_statistics_datawriter_with_profile()
+
+    // NETWORK_LATENCY_TOPIC is not defined in the fastdds.statistics property policy,
+    // it is just defined as data_writer profile. Thus, should not be created
+    std::string network_latency_name = "_fastdds_statistics_network_latency";
+    eprosima::fastdds::dds::DataWriter* network_latency_writer =
+            statistics_publisher_impl->lookup_datawriter(network_latency_name);
+    ASSERT_EQ(network_latency_writer, nullptr);
+
+    // But user can enable it manualy through enable_statistics_datawriter_with_profile()
+    ReturnCode_t ret = test_statistics_domain_participant_impl->enable_statistics_datawriter_with_profile(
+        "NETWORK_LATENCY_TOPIC");
+    ASSERT_EQ(ReturnCode_t::RETCODE_OK, ret);
+    network_latency_writer =
+            statistics_publisher_impl->lookup_datawriter(network_latency_name);
+    ASSERT_NE(network_latency_writer, nullptr);
+
+    efd::DataWriterQos qos3;
+    ASSERT_EQ(qos3, network_latency_writer->get_qos());
+
+    // SUBSCRIPTION_THROUGHPUT_TOPIC is not defined in the fastdds.statistics property policy,
+    // it is just defined as data_writer profile. Thus, should not be created
+    std::string subscription_througput_name = "_fastdds_statistics_subscription_throughput";
+    eprosima::fastdds::dds::DataWriter* subscription_througput_writer =
+            statistics_publisher_impl->lookup_datawriter(subscription_througput_name);
+    ASSERT_EQ(subscription_througput_writer, nullptr);
+
+    // But user can enable it manualy through enable_statistics_datawriter_with_profile()
+    ret = test_statistics_domain_participant_impl->enable_statistics_datawriter_with_profile(
+        "SUBSCRIPTION_THROUGHPUT_TOPIC");
+    ASSERT_EQ(ReturnCode_t::RETCODE_OK, ret);
+    subscription_througput_writer =
+            statistics_publisher_impl->lookup_datawriter(subscription_througput_name);
+    ASSERT_NE(subscription_througput_writer, nullptr);
+
+    // Expected QoS construction for Subscription_Throughput topic:
+    efd::DataWriterQos qos4;
+    qos4.reliability().kind = eprosima::fastdds::dds::ReliabilityQosPolicyKind::BEST_EFFORT_RELIABILITY_QOS;
+    eprosima::fastrtps::rtps::Property property;
+    property.name("partitions");
+    std::string partitions = "part1;part2";
+    property.value(std::move(partitions));
+    qos4.properties().properties().push_back(std::move(property));
+    qos4.deadline().period.seconds = 3;
+    qos4.deadline().period.nanosec = 0; // If seconds are set in XML, nanoseconds are set to 0 (while default value is max value)
+    qos4.latency_budget().duration.seconds = 2;
+    qos4.reliable_writer_qos().disable_heartbeat_piggyback = true;
+    ASSERT_EQ(qos4, subscription_througput_writer->get_qos());
+
```

Comment:
I am missing a couple of test cases:

1. Calling `enable_statistics_datawriter_with_profile` with a profile that does not exist. `RETCODE_ERROR` must be returned.
2. Loading a non consistent XML profile. This test case could be interesting of testing both with a builtin statistics datawriter as with the API that creates them manually.

Reply:
Done:
964ea3d2d Refs #15014: Test case for non existant topic.
c34fc2659 Refs #15014: Test case for DataWriters with non consistent QoS.